packages/terra-framework-docs/src/terra-dev-site/doc/compact-interactive-list/Examples.2/CellContent.jsx

```diff
@@ -0,0 +1,143 @@
+
+const updateRows = (rowsToUpdate, columns, selectionDetails) => {
+  const columnIndex = columns.findIndex(column => selectionDetails?.columnId === column.id);
+  return [...rowsToUpdate.map(row => {
+    const newRow = {
+      ...row,
+      cells: [...row.cells.map((cell, index) => {
+        const isSelectedCell = selectionDetails?.rowId === row.id && index === columnIndex;
+        // eslint-disable-next-line no-nested-ternary
+        const isSelected = isSelectedCell ? !cell.isSelected : false;
+        let newContent = cell.content;
+        if (index === 3) {
+          newContent = isSelected ? iconFeatured : iconFeaturedOff;
+        }
+        return { ...cell, isSelected, content: newContent };
+      })],
+    };
+    return newRow;
+  })];
+};
+
+const rows = [
+  {
+    id: 'row_1',
+    cells: [
+      { content: iconResultsNormal },
+      { content: 'Discern Care Set (1)' },
+      { content: buttonCell },
+      { content: iconFeaturedOff },
+    ],
+  },
+  {
+    id: 'row_2',
+    cells: [
+      { content: iconResultsNormal },
+      { content: 'Initial observation Care/Day High Severity 99220 (2)' },
+      { content: anchorCell },
+      { content: iconFeaturedOff },
+    ],
+  },
+  {
+    id: 'row_3',
+    cells: [
+      { content: iconResultsNotNormal },
+      { content: 'Arterial Sheath Care (3)' },
+      { content: anchorCell },
+      { content: iconFeaturedOff },
+    ],
+  },
+  {
+    id: 'row_4',
+    cells: [
+      { content: ' ' },
+      { content: 'Sbsq Observation Care/Day High Severity 99226 (4)' },
+      { content: buttonCell },
+      { content: iconFeaturedOff },
+    ],
+  },
+  {
+    id: 'row_5',
+    cells: [
+      { content: iconResultsCritical },
+      { content: 'Arterial Sheath Care (5)' },
+      { content: anchorCell },
+      { content: iconFeaturedOff },
+    ],
+  },
+];
+
+const cols = [
+  {
+    id: 'Column-0',
+    displayName: 'Icon',
+    width: '50px',
+    align: alignTypes.CENTER,
+    isSelectable: true,
+  },
+  {
+    id: 'Column-1',
+    displayName: 'Service name',
+    width: '250px', // will be used as a css flexBasis
+    flexGrow: true, // makes the column grow or shrink
+    maximumWidth: '350px',
+    minimumWidth: '100px',
+    isSelectable: true,
+  },
+  {
+    id: 'Column-2',
+    displayName: 'Details',
+    width: '150px',
+    align: alignTypes.RIGHT,
+    isSelectable: true,
+  },
+  {
+    id: 'Column-3',
+    displayName: 'Featured',
+    width: '50px',
+    align: alignTypes.CENTER,
+    isSelectable: true,
+  },
+];
+
+const CellContent = () => {
+  const [displayedRows, setDisplayedRows] = useState(updateRows(rows, cols));
+
+  const onCellSelect = ({ rowId, columnId }) => {
+    setDisplayedRows(updateRows(displayedRows, cols, { rowId, columnId }));
+  };
+
+  const onClearSelection = () => {
+    setDisplayedRows(updateRows(rows, cols));
+  };
```

Comment:
You should be using the useCallback hook for these methods.  Same comment for any method passed in your other examples.  Your current implementation will cause any component accepting the method as a prop to rerender even if memoized.

Reply:
Good catch, added in https://github.com/cerner/terra-framework/pull/1950/commits/a05da56ab53470a876d0d2de33585afa8fa78a50

The rest of examples don't have methods, as they don't have onCellSelect method passed